rxnetty-common/src/main/java/io/reactivex/netty/client/pool/PooledConnectionProviderImpl.java

```diff
@@ -70,18 +70,18 @@ public PooledConnectionProviderImpl(PoolConfig<W, R> poolConfig, HostConnector<W
         limitDeterminationStrategy = poolConfig.getPoolLimitDeterminationStrategy();
         maxIdleTimeMillis = poolConfig.getMaxIdleTimeMillis();
         // In case, there is no cleanup required, this observable should never give a tick.
-        idleConnCleanupSubscription = poolConfig.getIdleConnectionsCleanupTimer()
-                                                .doOnError(LogErrorAction.INSTANCE)
-                .retry() // Retry when there is an error in timer.
-                .concatMap(new IdleConnectionCleanupTask())
-                .onErrorResumeNext(new Func1<Throwable, Observable<Void>>() {
-                    @Override
-                    public Observable<Void> call(Throwable throwable) {
-                        logger.error("Ignoring error cleaning up idle connections.",
-                                     throwable);
-                        return Observable.empty();
-                    }
-                }).subscribe(Actions.empty()); // Errors are logged and ignored.
+        idleConnCleanupSubscription = poolConfig.getIdleConnCleanupTicker()
+            .doOnError(LogErrorAction.INSTANCE)
+            .retry() // Retry when there is an error in timer.
+            .concatMap(new IdleConnectionCleanupTask())
+            .onErrorResumeNext(new Func1<Throwable, Observable<Void>>() {
```

Comment:
Looking at this, I think the `onErrorResumeNext(...)` should be replaced by a `doOnError(...)` to log the error, then a `retry()`. 
The current change fixes the problem of the pool only being cleaned up once, but if there is a transient failure then it'll stop cleaning up.

Reply:
@jamesgorman2 the first commit is obsolete when i found the timer - interval problem but I forgot to revert. Will push revert of that commit.

Reply:
@jamesgorman2 sorry, I now see I did revert back to the original state of the subscription. And the onErrorResumeNext will log the error and then return empty without error so the cleaning up will not stop right? Or how do you mean by transient error?

Reply:
@rassmate transient probably refers to errors originating from IdleConnectionCleanupTask. An error is a terminal event in Rx so the stream will be aborted. A retry would subscribe the stream again.